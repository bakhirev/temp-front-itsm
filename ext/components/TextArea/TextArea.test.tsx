import React from 'react';
import { shallow } from 'enzyme';

import { TextArea } from './';

const testText =
  'С другой стороны рамки и место обучения кадров представляет собой интересный эксперимент проверки модели развития. Повседневная практика показывает, что консультация с широким активом обеспечивает широкому кругу (специалистов) участие в формировании форм развития. Идейные соображения высшего порядка, а также сложившаяся структура организации в значительной степени обуславливает создание соответствующий условий активизации. Таким образом консультация с широким активом представляет собой интересный эксперимент проверки систем массового участия.';

jest.mock('../common', () => ({
  DEFAULT_THEME: {},
}));

const handleOnChange = jest.fn();

describe('TextArea', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render component basically', () => {
    const wrapper = shallow(<TextArea />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with label', () => {
    const wrapper = shallow(<TextArea label="Label" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with limit information', () => {
    const wrapper = shallow(<TextArea limit={100} initialValue="Small text" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with label and limit information', () => {
    const wrapper = shallow(<TextArea label="Label" limit={100} initialValue="Small text" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with scrollbar and in default state', () => {
    const wrapper = shallow(<TextArea status={'default'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with error state', () => {
    const wrapper = shallow(<TextArea limit={100} initialValue={testText} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render disabled component', () => {
    const wrapper = shallow(<TextArea disabled initialValue={testText} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render readonly component', () => {
    const wrapper = shallow(<TextArea readOnly initialValue={testText} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should react on value change and call onChange callback', () => {
    const wrapper = shallow(
      <TextArea
        elementClassname="test-textarea"
        onChange={handleOnChange}
        initialValue={testText}
      />
    );
    const textarea = wrapper.find('.test-textarea');
    textarea.simulate('change', { target: { value: 'Text' } });
    expect(handleOnChange).toBeCalledWith({ target: { value: 'Text' } });
  });
});
