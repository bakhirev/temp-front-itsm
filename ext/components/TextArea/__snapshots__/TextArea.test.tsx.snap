// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea should render component basically 1`] = `
<styled.div
  $focused={false}
  $width="320px"
  theme={Object {}}
>
  <Styled(TextareaAutosize)
    $focused={false}
    $resize="none"
    $status="default"
    inputRef={
      Object {
        "current": null,
      }
    }
    maxLength={999}
    maxRows={10}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    theme={Object {}}
    value=""
  />
  <styled.div>
    <Styled(styled.div)
      $status="default"
      theme={Object {}}
    />
  </styled.div>
</styled.div>
`;

exports[`TextArea should render component with error state 1`] = `
<styled.div
  $focused={false}
  $width="320px"
  theme={Object {}}
>
  <Styled(TextareaAutosize)
    $focused={false}
    $resize="none"
    $status="error"
    inputRef={
      Object {
        "current": null,
      }
    }
    maxLength={100}
    maxRows={10}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    theme={Object {}}
    value="С другой стороны рамки и место обучения кадров представляет собой интересный эксперимент проверки модели развития. Повседневная практика показывает, что консультация с широким активом обеспечивает широкому кругу (специалистов) участие в формировании форм развития. Идейные соображения высшего порядка, а также сложившаяся структура организации в значительной степени обуславливает создание соответствующий условий активизации. Таким образом консультация с широким активом представляет собой интересный эксперимент проверки систем массового участия."
  />
  <styled.div>
    <Styled(styled.div)
      $status="error"
      theme={Object {}}
    >
      548
       / 
      100
    </Styled(styled.div)>
  </styled.div>
</styled.div>
`;

exports[`TextArea should render component with label 1`] = `
<styled.div
  $focused={false}
  $width="320px"
  theme={Object {}}
>
  <styled.div>
    Label
  </styled.div>
  <styled.div />
  <Styled(TextareaAutosize)
    $focused={false}
    $resize="none"
    $status="default"
    inputRef={
      Object {
        "current": null,
      }
    }
    maxLength={999}
    maxRows={10}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    theme={Object {}}
    value=""
  />
  <styled.div>
    <Styled(styled.div)
      $status="default"
      theme={Object {}}
    />
  </styled.div>
</styled.div>
`;

exports[`TextArea should render component with label and limit information 1`] = `
<styled.div
  $focused={false}
  $width="320px"
  theme={Object {}}
>
  <styled.div>
    Label
  </styled.div>
  <styled.div />
  <Styled(TextareaAutosize)
    $focused={false}
    $resize="none"
    $status="default"
    inputRef={
      Object {
        "current": null,
      }
    }
    maxLength={100}
    maxRows={10}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    theme={Object {}}
    value="Small text"
  />
  <styled.div>
    <Styled(styled.div)
      $status="default"
      theme={Object {}}
    >
      10
       / 
      100
    </Styled(styled.div)>
  </styled.div>
</styled.div>
`;

exports[`TextArea should render component with limit information 1`] = `
<styled.div
  $focused={false}
  $width="320px"
  theme={Object {}}
>
  <Styled(TextareaAutosize)
    $focused={false}
    $resize="none"
    $status="default"
    inputRef={
      Object {
        "current": null,
      }
    }
    maxLength={100}
    maxRows={10}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    theme={Object {}}
    value="Small text"
  />
  <styled.div>
    <Styled(styled.div)
      $status="default"
      theme={Object {}}
    >
      10
       / 
      100
    </Styled(styled.div)>
  </styled.div>
</styled.div>
`;

exports[`TextArea should render component with scrollbar and in default state 1`] = `
<styled.div
  $focused={false}
  $width="320px"
  theme={Object {}}
>
  <Styled(TextareaAutosize)
    $focused={false}
    $resize="none"
    $status="default"
    inputRef={
      Object {
        "current": null,
      }
    }
    maxLength={999}
    maxRows={10}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    theme={Object {}}
    value=""
  />
  <styled.div>
    <Styled(styled.div)
      $status="default"
      theme={Object {}}
    />
  </styled.div>
</styled.div>
`;

exports[`TextArea should render disabled component 1`] = `
<styled.div
  $focused={false}
  $width="320px"
  disabled={true}
  theme={Object {}}
>
  <Styled(TextareaAutosize)
    $focused={false}
    $resize="none"
    $status="default"
    disabled={true}
    inputRef={
      Object {
        "current": null,
      }
    }
    maxLength={999}
    maxRows={10}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    theme={Object {}}
    value="С другой стороны рамки и место обучения кадров представляет собой интересный эксперимент проверки модели развития. Повседневная практика показывает, что консультация с широким активом обеспечивает широкому кругу (специалистов) участие в формировании форм развития. Идейные соображения высшего порядка, а также сложившаяся структура организации в значительной степени обуславливает создание соответствующий условий активизации. Таким образом консультация с широким активом представляет собой интересный эксперимент проверки систем массового участия."
  />
  <styled.div>
    <Styled(styled.div)
      $status="default"
      theme={Object {}}
    />
  </styled.div>
</styled.div>
`;

exports[`TextArea should render readonly component 1`] = `
<styled.div
  $focused={false}
  $width="320px"
  theme={Object {}}
>
  <Styled(TextareaAutosize)
    $focused={false}
    $resize="none"
    $status="default"
    inputRef={
      Object {
        "current": null,
      }
    }
    maxLength={999}
    maxRows={10}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    readOnly={true}
    theme={Object {}}
    value="С другой стороны рамки и место обучения кадров представляет собой интересный эксперимент проверки модели развития. Повседневная практика показывает, что консультация с широким активом обеспечивает широкому кругу (специалистов) участие в формировании форм развития. Идейные соображения высшего порядка, а также сложившаяся структура организации в значительной степени обуславливает создание соответствующий условий активизации. Таким образом консультация с широким активом представляет собой интересный эксперимент проверки систем массового участия."
  />
  <styled.div>
    <Styled(styled.div)
      $status="default"
      theme={Object {}}
    />
  </styled.div>
</styled.div>
`;
